path_login.go

```diff
@@ -454,3 +462,47 @@ func convertPtrToString(s *string) string {
 	}
 	return ""
 }
+
+// getAPIVersionForResource attempts to query the supported API versions for a
+// given resource.
+func (b *azureAuthBackend) getAPIVersionForResource(ctx context.Context, subscriptionID, resourceID string) (string, error) {
+	client, err := b.provider.ProvidersClient(subscriptionID)
+	if err != nil {
+		return "", fmt.Errorf("unable to create providers client: %w", err)
+	}
+
+	elements := strings.Split(resourceID, "/")
+	if len(elements) < 9 {
+		return "", fmt.Errorf("unable to parse the resource ID: %s", resourceID)
+	}
+	providerNamespace := elements[6]
+	resourceType := elements[7]
```

Comment:
Microsoft provides a public API for parsing resource IDs [here](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/arm#ParseResourceID)